src/python_testing/TestConformanceTest.py

```diff
@@ -0,0 +1,131 @@
+        attrs[attr.ClusterRevision.attribute_id] = self.xml_clusters[Clusters.Descriptor.id].revision
+        attrs[attr.DeviceTypeList.attribute_id] = [
+            Clusters.Descriptor.Structs.DeviceTypeStruct(deviceType=device_type_id, revision=device_type_revision)]
+        attrs[attr.ServerList.attribute_id] = required_servers
+        attrs[attr.ClientList.attribute_id] = required_clients
+        attrs[attr.PartsList.attribute_id] = []
+        attrs[attr.AttributeList.attribute_id] = []
+        attrs[attr.AttributeList.attribute_id] = list(attrs.keys())
+
+        endpoint_tlv[Clusters.Descriptor.id] = attrs
+        return endpoint_tlv
+
+    def add_macl(self, root_endpoint: dict[int, dict[int, Any]]):
+        ac = Clusters.AccessControl
+        root_endpoint[ac.id][ac.Attributes.FeatureMap.attribute_id] = ac.Bitmaps.Feature.kManagedDevice
+        root_endpoint[ac.id][ac.Attributes.Arl.attribute_id] = []
+        root_endpoint[ac.id][ac.Attributes.CommissioningARL.attribute_id] = []
+        root_endpoint[ac.id][ac.Attributes.AttributeList.attribute_id].extend([
+            ac.Attributes.Arl.attribute_id, ac.Attributes.CommissioningARL.attribute_id])
+        root_endpoint[ac.id][ac.Attributes.AcceptedCommandList.attribute_id].append(ac.Commands.ReviewFabricRestrictions.command_id)
+        root_endpoint[ac.id][ac.Attributes.GeneratedCommandList.attribute_id].append(
+            ac.Commands.ReviewFabricRestrictionsResponse.command_id)
+
+    @async_test_body
+    async def test_macl_handling(self):
+        nim_id = self._get_device_type_id('network infrastructure manager')
+        root_node_id = self._get_device_type_id('root node')
+        on_off_id = self._get_device_type_id('On/Off Light')
+
+        root = self._create_minimal_dt(device_type_id=root_node_id)
+        nim = self._create_minimal_dt(device_type_id=nim_id)
+        self.endpoints_tlv = {0: root, 1: nim}
+        asserts.assert_true(self._has_nim(), "Did not find NIM in generated device")
+
+        success, problems = self.check_conformance(ignore_in_progress=False, is_ci=False)
+        self.problems.extend(problems)
+        asserts.assert_true(success, "Unexpected failure parsing minimal dt")
+
+        self.add_macl(root)
+        # A MACL is allowed when there is a NIM, so this should succeed as well
+        success, problems = self.check_conformance(ignore_in_progress=False, is_ci=False)
+        self.problems.extend(problems)
+        asserts.assert_true(success, "Unexpected failure with NIM and MACL")
+
+        # A MACL is not allowed when there is no NIM
+        self.endpoints_tlv[1] = self._create_minimal_dt(device_type_id=on_off_id)
+        success, problems = self.check_conformance(ignore_in_progress=False, is_ci=False)
+        self.problems.extend(problems)
+        asserts.assert_false(success, "Unexpected success with On/Off and MACL")
+
+        # TODO: what happens if there is a NIM and a non-NIM endpoint?
```

Comment:
That is allowed. The ARL entries would only be allowed on the clusters of the NIM endpoint

Reply:
Where in the spec is this stated? The ACL cluster is node scopes so it applies to all the EP's on the device and I have not found any such restrictions mentioned in the spec?